src/middlewared/middlewared/plugins/pool_/import_disk.py

```diff
@@ -1,20 +1,85 @@
-import asyncio
-import async_timeout
 import os
 import logging
 import re
 import subprocess
+import time
+
+import psutil
 
 from middlewared.job import JobProgressBuffer
-from middlewared.schema import Dict, returns, Str
-from middlewared.service import accepts, CallError, job, Service
-from middlewared.utils import Popen
+from middlewared.schema import accepts, Dict, List, returns, Str
+from middlewared.service import CallError, job, Service
 
 logger = logging.getLogger(__name__)
 
 
+def is_mounted(path):
+    mounted = psutil.disk_partitions()
+    return any(fs.mountpoint == path for fs in mounted)
+
+
+def mount(device, path, fs_type, fs_options, options):
+    options = options or []
+
+    if isinstance(device, str):
+        device = device.encode("utf-8")
+
+    if isinstance(path, str):
+        path = path.encode("utf-8")
+
+    executable = "mount"
+    arguments = []
+
+    if fs_type == "msdosfs" and fs_options:
+        if fs_options.get("locale"):
+            if fs_options.get("locale") == "utf8":
+                options.append("utf8")
+            else:
+                options.append(f"iocharset={fs_options['locale']}")
+
+    arguments.extend(["-t", {"msdosfs": "vfat", "ext2fs": "ext2"}.get(fs_type, fs_type)])
+
+    if options:
+        arguments.extend(["-o", ",".join(options)])
+
+    proc = subprocess.Popen(
+        [executable] + arguments + [device, path],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    output = proc.communicate()
+
+    if proc.returncode != 0:
+        logger.debug("Mount failed (%s): %s", proc.returncode, output)
+        raise ValueError("Mount failed (exit code {0}):\n{1}{2}" .format(
+            proc.returncode,
+            output[0].decode("utf-8"),
+            output[1].decode("utf-8"),
+        ))
+    else:
+        return True
+
+
+class MountFsContextManager:
```

Comment:
Wouldn't it make more sense to have the mountpoint creation and deletion be part of the context manager?